x-pack/plugins/enterprise_search/server/routes/app_search/search_settings.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+import { RouteDependencies } from '../../plugin';
+
+// We only do a very light type check here, and allow unknowns, because the request is validated
+// on the ent-search server, so it would be redundant to check it here as well.
+const boosts = schema.recordOf(
+  schema.string(),
+  schema.arrayOf(schema.object({}, { unknowns: 'allow' }))
+);
+const resultFields = schema.recordOf(schema.string(), schema.object({}, { unknowns: 'allow' }));
+const searchFields = schema.recordOf(schema.string(), schema.object({}, { unknowns: 'allow' }));
+
+const searchSettingsSchema = schema.object({
+  boosts,
+  result_fields: resultFields,
+  search_fields: searchFields,
+});
+
+export function registerSearchSettingsRoutes({
+  router,
+  enterpriseSearchRequestHandler,
+}: RouteDependencies) {
+  router.get(
+    {
+      path: '/api/app_search/engines/{engineName}/search_settings/details',
+      validate: {
+        params: schema.object({
+          engineName: schema.string(),
+        }),
+      },
+    },
+    async (context, request, response) => {
+      return enterpriseSearchRequestHandler.createRequest({
+        path: `/as/engines/${request.params.engineName}/search_settings/details`,
+      })(context, request, response);
```

Comment:
Scotty made me realize that when we're not modifying the context/request/response, we don't need to wrap this in an async function:

```
enterpriseSearchRequestHandler.createRequest({
  path: '/as/engines',
})
```

is equivalent.  If you wanted you could use this everywhere in this file and save 8 lines!

Reply:
In some cases that IS true, but when you need to set a path parameter I believe that is not the. I think the handler requires that you give it a full path, I don't think you can just give it a base path. And since the path requires path parameter, this wrapper is required in order to extract engineName and build the path.

Reply:
c779f6a